src/overlays/actors/ovl_En_Goma/z_en_goma.c

```diff
@@ -73,77 +104,796 @@ static ColliderCylinderInit D_80A4B7CC = {
+    }
+}
+
+void EnGoma_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnGoma* this = THIS;
+    s32 pad;
+    Player* player = PLAYER;
+
+    if (this->actionTimer != 0) {
+        this->actionTimer--;
+    }
+    if (this->invincibilityTimer != 0) {
+        this->invincibilityTimer--;
+    }
+
+    this->actionFunc(this, globalCtx);
+    Actor_MoveForward(thisx);
+    this->actor.world.pos.x = this->actor.world.pos.x + this->shieldKnockbackVel.x;
+    this->actor.world.pos.z = this->actor.world.pos.z + this->shieldKnockbackVel.z;
+    Math_ApproachZeroF(&this->shieldKnockbackVel.x, 1.0f, 3.0f);
+    Math_ApproachZeroF(&this->shieldKnockbackVel.z, 1.0f, 3.0f);
+
+    if (this->actor.params < 10) {
+        this->eggTimer++;
+        Math_SmoothStepToF(&this->actor.scale.x, 0.01f, 0.5f, 0.00075f, 0.000001f);
+        Math_SmoothStepToF(&this->actor.scale.y, 0.01f, 0.5f, 0.00075f, 0.000001f);
+        Math_SmoothStepToF(&this->actor.scale.z, 0.01f, 0.5f, 0.00075f, 0.000001f);
+        EnGoma_UpdateHit(this, globalCtx);
+        Actor_UpdateBgCheckInfo(globalCtx, thisx, 50.0f, 50.0f, 100.0f, 5);
+        EnGoma_SetFloorRot(this);
+        Actor_SetFocus(thisx, 20.0f);
+        EnGoma_LookAtPlayer(this, globalCtx);
+        EnGoma_UpdateEyeEnvColor(this);
+        this->visualState = 1;
+        if (player->swordState != 0) {
+            this->colCyl2.dim.radius = 35;
+            this->colCyl2.dim.height = 35;
+            this->colCyl2.dim.yShift = 0;
+        } else {
+            this->colCyl2.dim.radius = 15;
+            this->colCyl2.dim.height = 30;
+            this->colCyl2.dim.yShift = 10;
+        }
+        if (this->invincibilityTimer == 0) {
+            Collider_UpdateCylinder(thisx, &this->colCyl1);
+            Collider_UpdateCylinder(thisx, &this->colCyl2);
+            CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->colCyl1.base);
+            CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colCyl2.base);
+            CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->colCyl1.base);
+        }
+    }
+}
+
+s32 EnGoma_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, void* thisx) {
+    EnGoma* this = THIS;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_goma.c", 1976);
+    gDPSetEnvColor(POLY_OPA_DISP++, (s16)this->eyeEnvColor[0], (s16)this->eyeEnvColor[1], (s16)this->eyeEnvColor[2],
+                   255);
+
+    if (limbIndex == 7) {
+        rot->x += this->eyePitch;
+        rot->y += this->eyeYaw;
+    } else if (limbIndex == 3 && this->hurtTimer != 0) {
+        gDPSetEnvColor(POLY_OPA_DISP++, (s16)(Rand_ZeroOne() * 255.0f), (s16)(Rand_ZeroOne() * 255.0f),
+                       (s16)(Rand_ZeroOne() * 255.0f), 255);
+    }
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_goma.c", 2011);
+    return 0;
+}
+
+Gfx* EnGoma_NoBackfaceCullingDlist(GraphicsContext* gfxCtx) {
+    Gfx* dl;
+    Gfx* dlStart;
```

Comment:
```suggestion
    Gfx* dList;
    Gfx* dListHead;
```
Is what we usually call these.